addons/pos_self_order/static/src/app/self_order_service.js

```diff
@@ -319,6 +319,11 @@ export class SelfOrder extends Reactive {
             product.pos_categ_ids.map((categ) => {
                 acc[categ.id] = acc[categ.id] || [];
                 acc[categ.id].push(product);
+                while (categ.parent_id) {
+                    acc[categ.parent_id[0]] = acc[categ.parent_id[0]] || [];
+                    acc[categ.parent_id[0]].push(product);
+                    categ = categ.parent_id;
+                }
```

Comment:
i don't really get this from a functional pov.
Why does the `self_order` care about category parents? The category selector in `pos` shows the parent/child relation between categories, but the one in self doesn't, so why would you need this code?

Reply:
Dama asked me to add it